deployments/pulse3d/services/pulse3d/src/main.py

```diff
@@ -763,9 +774,8 @@ async def get_versions(request: Request):
     try:
         async with request.state.pgpool.acquire() as con:
             rows = await con.fetch(  # TODO should eventually sort these using a more robust method
-                "SELECT version, state FROM pulse3d_versions WHERE state != 'deprecated' ORDER BY created_at"
+                "SELECT version, state, end_of_life_date FROM pulse3d_versions WHERE end_of_life_date  > NOW() OR end_of_life_date IS null   ORDER BY created_at"
```

Comment:
https://github.com/CuriBio/k8s_curi/pull/437#discussion_r1136106757

Reply:
`SELECT version, state, end_of_life_date FROM pulse3d_versions WHERE end_of_life_date  > NOW() OR end_of_life_date IS null   ORDER BY created_at"` works.  I was able to submit an analysis.

Reply:
It may, but it's less clear about what it's doing. Getting code to work is only half the picture, we also need to make sure that other people are able to easily understand and maintain code. This query makes no mention of deprecated states, so someone who doesn't know or forgets that we will only have the EOL column set for deprecated versions will have a harder time understanding the intention of this query

Reply:
I can add a comment to make the query more clear. 
This is the final query `"SELECT version, state, end_of_life_date FROM pulse3d_versions WHERE state != 'deprecated' OR NOW() < end_of_life_date ORDER BY created_at"` and it does not fetch deprecated versions ever. If a version is marked deprecated it will never be fetched for the user. We want to only compare the end_of_life_date and see if it passed. Adding state != deprecated would just make it so versions marked with deprecated will never appear for the user.  So I think the current version is better because it check if the end of life date has passed.

Reply:
![Capture](https://user-images.githubusercontent.com/46114925/225749326-e68ed711-0000-4470-a1e1-dafdb549dd64.PNG)

Reply:
This is what it looks like with your suggested query.